packages/framework/src/revert-reason.ts

```diff
@@ -1,15 +1,18 @@
-import { abi } from 'thor-devkit'
+import { abi, keccak256 } from 'thor-devkit'
 
-// https://solidity.readthedocs.io/en/v0.5.5/control-structures.html#error-handling-assert-require-revert-and-exceptions
-// 0x08c379a0
-// Function selector for Error(string)
+// https://docs.soliditylang.org/en/v0.8.16/control-structures.html#error-handling-assert-require-revert-and-exceptions
+// builtin errors in solidity, Error(string) and Panic(uint256)
 
-const errorSig = '0x08c379a0'
+const errorSelector = '0x' + keccak256('Error(string)').toString('hex').slice(0, 8)
+const panicSelector = '0x' + keccak256('Panic(uint256)').toString('hex').slice(0, 8)
 
 export function decodeRevertReason(data: string): string {
     try {
-        if (data.startsWith(errorSig)) {
-            return abi.decodeParameter('string', '0x' + data.slice(errorSig.length)) as string
+        if (data.startsWith(errorSelector)) {
+            return abi.decodeParameter('string', '0x' + data.slice(errorSelector.length)) as string
+        } else if (data.startsWith(panicSelector)) {
+            const decoded = abi.decodeParameter('uint256', '0x' + data.slice(panicSelector.length)) as string
+            return `Panic(0x${parseInt(decoded).toString(16).padStart(2, '0')})`
```

Comment:
Will `decoded` be too long for parseInt?

Reply:
No, the max value is 0x51. It is defined inside the solidity compiler:

```
0x00: Used for generic compiler inserted panics.

0x01: If you call assert with an argument that evaluates to false.

0x11: If an arithmetic operation results in underflow or overflow outside of an unchecked { ... } block.

0x12; If you divide or modulo by zero (e.g. 5 / 0 or 23 % 0).

0x21: If you convert a value that is too big or negative into an enum type.

0x22: If you access a storage byte array that is incorrectly encoded.

0x31: If you call .pop() on an empty array.

0x32: If you access an array, bytesN or an array slice at an out-of-bounds or negative index (i.e. x[i] where i >= x.length or i < 0).

0x41: If you allocate too much memory or create an array that is too large.

0x51: If you call a zero-initialized variable of internal function type.
```

Reply:
https://docs.soliditylang.org/en/v0.8.16/control-structures.html#panic-via-assert-and-error-via-require

Reply:
👍